geomesa-core/src/main/scala/org/locationtech/geomesa/core/transform/TransformCreator.scala

```diff
@@ -18,27 +19,25 @@ object TransformCreator {
    */
   def createTransform(targetFeatureType: SimpleFeatureType,
                       featureEncoding: FeatureEncoding,
-                      transformString: String): (SimpleFeature => Array[Byte]) =
+                      transformString: String): (SimpleFeature => Array[Byte]) = {
+
+    val encoder = SimpleFeatureEncoder(targetFeatureType, featureEncoding)
+    val defs = TransformProcess.toDefinition(transformString)
     featureEncoding match {
-      case FeatureEncoding.AVRO =>
-        val encoder = SimpleFeatureEncoder(targetFeatureType, featureEncoding)
-        val defs = TransformProcess.toDefinition(transformString)
+      case FeatureEncoding.KRYO | FeatureEncoding.AVRO =>
         (feature: SimpleFeature) => {
-          val newSf = new AvroSimpleFeature(feature.getIdentifier, targetFeatureType)
+          val newSf = new KryoSimpleFeature(feature.getIdentifier.getID, targetFeatureType)
           defs.foreach { t => newSf.setAttribute(t.name, t.expression.evaluate(feature)) }
           encoder.encode(newSf)
         }
 
       case FeatureEncoding.TEXT =>
-        val defs = TransformProcess.toDefinition(transformString)
-        val encoder = SimpleFeatureEncoder(targetFeatureType, featureEncoding)
-        val builder = AvroSimpleFeatureFactory.featureBuilder(targetFeatureType)
+        val builder = KryoSimpleFeatureFactory.featureBuilder(targetFeatureType)
         (feature: SimpleFeature) => {
-          builder.reset()
```

Comment:
SimpleFeatureBuilder.buildFeature() calls reset internally (line 356), so the call is redundant...